controllers/bsl.go

```diff
@@ -95,8 +95,40 @@ func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool,
 				Namespace: r.NamespacedName.Namespace,
 			},
 		}
+		// Add label oadpApi.OadpOperatorLabel to the bsl secret
+		// which in turn will add it to the reconciliation loop
+
+		secretName, _ := r.getSecretNameAndKey(bslSpec.Velero.Credential, oadpv1alpha1.DefaultPlugin(bslSpec.Velero.Provider))
+		secret, _ := r.getProviderSecret(secretName)
+		sec := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: r.NamespacedName.Namespace,
+			},
+		}
+		secret.SetLabels(map[string]string{oadpv1alpha1.OadpOperatorLabel: "True", "dpaName": dpa.Name, "namespace": dpa.Namespace})
```

Comment:
Just curious do we need to be setting the namespace label? All these secrets should be in the same namespace as the operator so feels redundant to me.

NIT: I would also change `dpaName` to `dataprotectionapplication: <name>`.

Reply:
When I populated the namespace from the secret object it did not work. The event never kicked off 🤷‍♀️  So I ended up adding a label to work around. @dymurray do you think its hacky? I think it is but I dont know any another way to get it working

Reply:
I am going to test it again. I changed a whole bunch of things and this might work again.

Reply:
I must have done something wrong yesterday. It works now. Thanks @dymurray for catching it :)